docs: "Manual integration releases from this repository to npm" and "Create local branches for fork branches from PRs" (#4406)
prisma/prisma-engines: README.md

When it's time to merge the sibling PRs, you'll need to merge the prisma/prisma PR first, so when merging the engines PR you have the code of the adapters ready in prisma/prisma `main` branch.

### Testing engines in `prisma/prisma`

> You can trigger releases from this repository to npm that can be used for testing the engines in `prisma/prisma` both automatically, or manually:

Reviewer: ```suggestion
You can trigger releases from this repository to npm that can be used for testing the engines in `prisma/prisma` either automatically or manually:
```

#### Automated integration releases from this repository to npm

> (Since July 2022). Any branch name starting with `integration/` will, first, run the full test suite in Buildkite `[Test] Prisma Engines` and, second, if passing, run the publish pipeline (build and upload engines to S3)

Reviewer: ```suggestion
(Since July 2022). Any branch name starting with `integration/` will, first, run the full test suite in Buildkite `[Test] Prisma Engines` and, second, if passing, run the publish pipeline (build and upload engines to S3 & R2)
```

- It will trigger [`prisma/engines-wrapper`](https://github.com/prisma/engines-wrapper) and publish a new [`@prisma/engines-version`](https://www.npmjs.com/package/@prisma/engines-version) npm package but on the `integration` tag.
- Which triggers [`prisma/prisma`](https://github.com/prisma/prisma) to create a `chore(Automated Integration PR): [...]` PR with a branch name also starting with `integration/`
- Since in `prisma/prisma` we also trigger the publish pipeline when a branch name starts with `integration/`, this will publish all `prisma/prisma` monorepo packages to npm on the `integration` tag.
- Our [ecosystem-tests](https://github.com/prisma/ecosystem-tests/) tests will automatically pick up this new version and run tests, results will show in [GitHub Actions](https://github.com/prisma/ecosystem-tests/actions?query=branch%3Aintegration)

This end to end will take minimum ~1h20 to complete, but is completely automated :robot:

Notes:
- in `prisma/prisma` repository, we do not run tests for `integration/` branches, it is much faster and also means that there is no risk of tests failing (e.g. flaky tests, snapshots) that would stop the publishing process.

> - in `prisma/prisma-engines` tests must first pass, before publishing starts. So better keep an eye on them and restart them as needed.

Reviewer: ```suggestion
- in `prisma/prisma-engines` the Buildkite test pipeline must first pass, then the engines will be built and uploaded to our storage in the Buildkite release pipeline. These 2 pipeline can fail for different reason, it's recommended to keep an eye on them (check notifications in Slack) and restart jobs as needed. Finally it will trigger [`prisma/engines-wrapper`](https://github.com/prisma/engines-wrapper). 
```

--target-dir:/tmp/rust-analyzer-check
```

## Community PRs: create a local branch for a branch coming from a fork

To trigger an [Automated integration releases from this repository to npm](#automated-integration-releases-from-this-repository-to-npm) or [Manual integration releases from this repository to npm](#manual-integration-releases-from-this-repository-to-npm) branches of forks need to be pulled into this repository so the Buildkite job is triggered. You can use these GitHub and git CLI commands to achieve that easily:

```
gh pr checkout 4375
git checkout -b integration/sql-nested-transactions
git push --set-upstream origin integration/sql-nested-transactions
```

> ## Security

Reviewer: ```suggestion
If there is a need to re-create this branch because it has been updated, deleting it and re-creating will make sure the content is identical.

```
git branch --delete integration/sql-nested-transactions
gh pr checkout 4375
git checkout -b integration/sql-nested-transactions
git push --set-upstream origin integration/sql-nested-transactions --force
```

## Security
```

Author: You can just change the outer triple backticks to four backticks and then it works :shrug:

--target-dir:/tmp/rust-analyzer-check
```

> ## Create local branches for fork branches from PRs

Reviewer: ```suggestion
## Community PRs: create a local branch for a branch coming from a fork
```

To trigger an [Automated integration releases from this repository to npm](#automated-integration-releases-from-this-repository-to-npm) or [Manual integration releases from this repository to npm](#manual-integration-releases-from-this-repository-to-npm) branches of forks need to be pulled into this repository so the Buildkite job is triggered. You can use these GitHub and git CLI commands to achieve that easily:

```
gh pr checkout 4375

> git branch integration/sql-nested-transactions
> git checkout integration/sql-nested-transactions

Reviewer: ```suggestion
git checkout -b integration/sql-nested-transactions
```
better as one line

> git push (something)

Reviewer: ```suggestion
git push --set-upstream origin integration/sql-nested-transactions
```

> git push --set-upstream origin integration/sql-nested-transactions

Author: We could turn that into a small script in the repo that takes the PR and the desired git branch and automates this.
